### PR TITLE
Add 6.8 branch to packer cache script

### PR DESCRIPTION
**Pull request by breskeby**
This allows us to cache dependencies from the elasticsearch 6.8 branch which should help reducing the overall CI failures on that branch caused by dependency resolution errors.

### CODE REVIEW

File: .ci/packer_cache.sh [hunk 1]
```diff
@@ -16,12 +16,21 @@ while [ -h "$SCRIPT" ] ; do
...
 export JAVA11_HOME="${HOME}"/.java/java11
 export JAVA12_HOME="${HOME}"/.java/openjdk12
 export JAVA13_HOME="${HOME}"/.java/openjdk13
 export JAVA14_HOME="${HOME}"/.java/openjdk14
-./gradlew --parallel clean -s resolveAllDependencies
 
+## 6.8 branch is not referenced from any bwc project in master so we need to
+## resolve its dependencies explicitly
+rm -rf checkout/6.8
+git clone https://github.com/elastic/elasticsearch.git --branch 6.8 --single-branch checkout/6.8
```
[the remark this answers is not in the document]
Comment (breskeby) [reply .ci/packer_cache.sh:29]: Nice. didn't know about that one.

**Review by nik9000 — APPROVED** [review]
LGTM. Not like I can test it locally but the clone bit works.

File: .ci/packer_cache.sh [hunk 2]
```diff
@@ -16,12 +16,21 @@ while [ -h "$SCRIPT" ] ; do
...
 export JAVA14_HOME="${HOME}"/.java/openjdk14
-./gradlew --parallel clean -s resolveAllDependencies
 
+## 6.8 branch is not referenced from any bwc project in master so we need to
+## resolve its dependencies explicitly
+rm -rf checkout/6.8
+git clone --reference $(dirname "${SCRIPT}")/../.git https://github.com/elastic/elasticsearch.git --branch 6.8 --single-branch checkout/6.8
+export JAVA_HOME="${JAVA11_HOME}"
+./checkout/6.8/gradlew --project-dir ./checkout/6.8 --parallel clean --scan -Porg.elasticsearch.acceptScanTOS=true --stacktrace resolveAllDependencies
+
```
Comment (nik9000) [inline .ci/packer_cache.sh:32]: Maybe `rm -rf checkout/6.8` just to keep the build image a little smaller.

**Review by mark-vieira — APPROVED** [review]
Brilliant :+1:

**Comment by mark-vieira** [comment]
FYI, I removed the `v7.11.0` label here since this only needs to go in `master` as that's  the only branch that our CI automation runs this script from.

File: .ci/packer_cache.sh (hunk 2 above)
Comment (breskeby) [reply .ci/packer_cache.sh]: added this

**Comment by breskeby** [comment]
@mark-vieira yeah forgot to remove the 7.x tag from my PR template before creating this. thx

**Pull request merged**